### PR TITLE
MS-24-Positions: positions features done

### PR DESCRIPTION
**Pull request by ManuelLopezAnido**
Positions done

**PR in server is needed!**

### CODE REVIEW

File: src/Components/Positions/positions.module.css [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .container {
-  flex: 1;
   display: flex;
+  flex-direction:column;
```
Comment (aquilesmariani) [inline src/Components/Positions/positions.module.css:3]:
```suggestion
  flex-direction: column;
```

File: src/Components/Positions/Form/FormPositions.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    if (PosId === null) {
+      options.method = 'POST';
+      url = `${process.env.REACT_APP_API}/api/positions/create`;
+    } else {
+      options.method = 'PUT';
+      url = `${process.env.REACT_APP_API}/api/positions/update/${PosId}`;
+    }
+    console.log("La url es: ", url); 
+    console.log("El curepo es: ", options);
```
Comment (aquilesmariani) [inline src/Components/Positions/Form/FormPositions.js:94]:
```suggestion
```

File: src/Components/Positions/Form/FormPositions.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        city: city,
+        country: country,
+        datePosted: datePosted,
+        closingDate: closingDate
+      })
+    };
+
+    if (PosId === null) {
+      options.method = 'POST';
+      url = `${process.env.REACT_APP_API}/api/positions/create`;
```
Comment (aquilesmariani) [inline src/Components/Positions/Form/FormPositions.js:88]: If you check on the deployed app in [heroku](https://basd-2021-aq-ms-24-posi-owlhvx.herokuapp.com/positions) , the requests to the server are failing cause `/api` is duplicated. It is probably that `/api` already included in the 
`REACT_APP_API` environment variable. You should remove it from here and add it in your `.env` file

```suggestion
      url = `${process.env.REACT_APP_API}/positions/create`;
```

File: .eslintrc.json [hunk 4]
```diff
@@ -1,27 +1,33 @@
...
+      "version": "detect"
     }
-}
+  },
+  "env": {
+    "browser": true,
+    "node": true,
+    "es2021": true
+  },
+  "extends": ["eslint:recommended", "plugin:react/recommended" /*, "plugin:prettier/recommended"*/],
```
Comment (aquilesmariani) [inline .eslintrc.json:12]: Let's roll back these file changes

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Please review comments I've also left [here](https://github.com/BaSD-2021/Aquiles-MindSet-app/pull/5#pullrequestreview-821036810) because most of them are repeated.

File: src/Components/Positions/index.js [hunk 5]
```diff
@@ -1,11 +1,92 @@
...
+    fetch(`${process.env.REACT_APP_API}/api/positions`)
+      .then((response) => response.json())
+      .then((response) => {
+        console.log(response);
+        setPositions(response);
+      });
+  }, []);
+  console.log(`${process.env.REACT_APP_API}/api/positions`);
+  console.log("datass: ", positions);
+
```
Comment (ignacioaristo) [inline src/Components/Positions/index.js:21]: Remove those console.log

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/Positions/index.js [hunk 6]
```diff
@@ -1,11 +1,90 @@
...
+import ModalPositons from './Modal/modalPositions.js';
+import deleteIcon from '../../assets/deleteIcon.png';
 
 function Positions() {
+  const [showModal, setShowModal] = useState(false);
+  const [positions, setPositions] = useState([]);
+  const [selectedId, setSelectedId] = useState('');
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API}/api/positions`)
```
Comment (ignacioaristo) [inline src/Components/Positions/index.js:12]: Maybe you need to remove this `/api`

File: src/Components/Positions/index.js [hunk 7]
```diff
@@ -1,11 +1,90 @@
...
+        setPositions(response);
+      });
+  }, []);
+ 
+  const addPositions = () =>{
+    window.location.href = `/positions/form`;
+  };
+
+  const deletePosition = (idPos) => {
+    const url = `${process.env.REACT_APP_API}/api/positions/delete/${idPos}`;
```
Comment (ignacioaristo) [inline src/Components/Positions/index.js:25]: This one too.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/Positions/Form/FormPositions.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        .then((response) => {
+          setJobTitle(response.jobTitle);
+          setClientId(response.clientId);
+          setCompanyName(response.companyName);
+          setjobDescription(response.jobDescription);
+          setCity(response.city);
+          setCountry(response.country);
+          setDatePosted(response.datePosted);
+          setClosingDate(response.closingDate);
+        });
```
Comment (leimeterjoaquin) [inline src/Components/Positions/Form/FormPositions.js:62]: add a catch.
maybe
```
.catch((error) => {
  return error;
});
```
This should be a modal, we can implement it for this week

**Pull request merged**